### DISCUSSION TITLE
Error - unable to reestablish IKE_SA due to asymmetric setup

### DISCUSSION THREAD

**Discussion by alex-airspan** [strongswan/strongswan#1806] [Q&A]
Hi 

I working on my client Strongswan with Route-based VPN with VTI Devices. Tunnel UP and traffic running. 
The SeGW also Strongswan. After some time that tunnel UP I see in SeGW  - creating rekey job with some retransmissions and after that error - unable to reestablish IKE_SA due to asymmetric setup

What it mean? how possible fix it ?

20-07-2023-12:44:28::09[KNL] creating rekey job for CHILD_SA ESP/0xc97dfd3c/172.27.123.20
20-07-2023-12:44:28::15[IKE] establishing CHILD_SA net{1084} reqid 1
20-07-2023-12:44:28::15[ENC] generating CREATE_CHILD_SA request 0 [ N(REKEY_SA) SA No TSi TSr ]
20-07-2023-12:44:28::15[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:44:32::01[IKE] retransmit 1 of request with message ID 0
20-07-2023-12:44:32::01[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:44:39::14[IKE] retransmit 2 of request with message ID 0
20-07-2023-12:44:39::14[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:44:52::01[IKE] retransmit 3 of request with message ID 0
20-07-2023-12:44:52::01[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:45:16::05[IKE] retransmit 4 of request with message ID 0
20-07-2023-12:45:16::05[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:45:27::01[KNL] creating rekey job for CHILD_SA ESP/0xcef9ae34/172.27.128.193
20-07-2023-12:45:44::
20-07-2023-12:45:58::12[IKE] retransmit 5 of request with message ID 0
20-07-2023-12:45:58::12[NET] sending packet: from 172.27.123.20[500] to 172.27.128.193[500] (208 bytes)
20-07-2023-12:46:22::
20-07-2023-12:47:13::01[KNL] creating delete job for CHILD_SA ESP/0xc8c02407/172.27.123.20
20-07-2023-12:47:13::01[JOB] CHILD_SA ESP/0xc8c02407/172.27.123.20 not found for delete
20-07-2023-12:47:13::05[IKE] giving up after 5 retransmits
20-07-2023-12:47:13::05[IKE] unable to reestablish IKE_SA due to asymmetric setup
20-07-2023-12:47:53::

Thanks,
Alex

**Comment by tobiasbrunner** [comment]
Check the config for asymmetric authentication settings (e.g. EAP) or virtual IP usage.

**Comment by alex-airspan** [comment]
Hi Tobias,

I set reauth_time parameters same in Server ans client. Now I not receive asymmetric setup message. 

But still issue with dpd request. I see some retransmissions and Data over tunnel stopped. 

dpd configuration: 
     dpd_action = restart
      dpd_delay = 30

05[IKE] sending DPD request
05[ENC] generating INFORMATIONAL request 120 [ ]
05[NET] sending packet: from 172.27.128.193[500] to 172.27.123.20[500] (80 bytes)
13[IKE] retransmit 1 of request with message ID 120
13[NET] sending packet: from 172.27.128.193[500] to 172.27.123.20[500] (80 bytes)
06[IKE] retransmit 2 of request with message ID 120
06[NET] sending packet: from 172.27.128.193[500] to 172.27.123.20[500] (80 bytes)
05[IKE] retransmit 3 of request with message ID 120
05[NET] sending packet: from 172.27.128.193[500] to 172.27.123.20[500] (80 bytes)
11[IKE] retransmit 4 of request with message ID 120
11[NET] sending packet: from 172.27.128.193[500] to 172.27.123.20[500] (80 bytes)
16[NET] received packet: from 172.27.123.20[500] to 172.27.128.193[500] (80 bytes)

How possible debug the dpd issue? 

Thanks,
Alex

**Reply by tobiasbrunner** [reply]
> I set reauth_time parameters same in Server ans client. Now I not receive asymmetric setup message.

That's not really a fix for that issue. While it could cause clients to reauthenticate (via AUTH_LIFETIME notify, maybe [read up on reauthentication](https://docs.strongswan.org/docs/5.9/config/rekeying.html#_ike_sas)), which wasn't even the case here, it does not change the fact that the server might not be able to reestablish the IKE_SA due to an asymmetric config if it isn't able to reach the client.

> How possible debug the dpd issue?

Check firewall, routing, network connectivity in general etc.

You wrote that you have a route-based setup. So maybe you cause some kind of routing loop (i.e. IKE packets are being sent via tunnel).

**Reply by tobiasbrunner** [reply]
> 16[NET] received packet: from 172.27.123.20[500] to 172.27.128.193[500] (80 bytes)

The log indicates that eventually there is a response. So maybe there is just too much other traffic for the DPDs (request or response) to get through.

**Comment by alex-airspan** [comment]
Hi Tobias,

I using according to your recommendation with below ip routing tables. 
SeGW - 172.27.123.20,   Client Public - 172.27.128.193

Maybe I mismatch anything. I'll check also with wireshark how sent DPD. 

[root@localhost ~]# ip route show table **main** 
default via 172.27.128.254 dev enp5s0f0 
172.27.128.0/24 dev enp5s0f0 proto kernel scope link src 172.27.128.193

[root@localhost ~]# ip route show table **220**
default dev ipsec0 proto static scope link 
throw 172.27.123.20 


[root@localhost ~]# ifconfig

enp5s0f0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.27.128.193  netmask 255.255.255.0  broadcast 172.27.128.255
        inet6 fe80::2a0:aff:fe00:5afc  prefixlen 64  scopeid 0x20<link>
        ether 00:a0:0a:00:5a:fc  txqueuelen 1000  (Ethernet)
        RX packets 62316471  bytes 89281487127 (83.1 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 23272991  bytes 3225924553 (3.0 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ipsec0: flags=209<UP,POINTOPOINT,RUNNING,NOARP>  mtu 1480
        inet 172.27.124.210  netmask 255.255.255.255  destination 172.27.124.210
        inet6 fe80::5efe:ac1b:80c1  prefixlen 64  scopeid 0x20<link>
        tunnel   txqueuelen 1000  (IPIP Tunnel)
        RX packets 62279320  bytes 84046992991 (78.2 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 23214927  bytes 1208312465 (1.1 GiB)
        TX errors 11  dropped 0 overruns 0  carrier 6  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 703723  bytes 328581452 (313.3 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 703723  bytes 328581452 (313.3 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

Thanks,
Alex
